internal/types/path.go

```diff
@@ -48,3 +49,78 @@ func getByPath[T CompositeTypeInterface](comp T, path ...string) (any, error) {
 
 	return next, nil
 }
+
+// getKeyPaths returns a key path.
+func getKeyPaths[T CompositeTypeInterface](comp T, key string, currentPath []string, in [][]string) (res [][]string, err error) {
```

Comment:
it will be used in compatibility with mongo driver, when `projection` filter might be in the `filter` document

Reply:
Projection is not a filter. Let's talk about it tomorrow

Reply:
removed this commit from this PR, let's make a separate PR for these details